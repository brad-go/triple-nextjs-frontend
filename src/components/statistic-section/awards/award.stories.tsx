import { ComponentMeta, ComponentStory } from '@storybook/react'

import Award from './award'

export default {
  title: 'statistic-section/award',
  component: Award,
} as ComponentMeta<typeof Award>

const Template: ComponentStory<typeof Award> = () => {
  return <Award />
}

export const Default = Template.bind({})

export const Mobile = Template.bind({})
Mobile.parameters = {
  viewport: {
    defaultViewport: 'mobile2',
  },
}
