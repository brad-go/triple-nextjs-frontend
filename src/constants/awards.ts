export interface AwardList {
  id: number
  platform: string
  award: string
  image: string
}

export const AWARD_LIST: AwardList[] = [
  {
    id: 1,
    platform: '2018 구글 플레이스토어',
    award: '올해의 앱 최우수상 수상',
    image: '/images/play-store2x.png',
  },
  {
    id: 2,
    platform: '2018 애플 앱스토어',
    award: '오늘의 여행앱 선정',
    image: '/images/badge-apple4x.png',
  },
]
