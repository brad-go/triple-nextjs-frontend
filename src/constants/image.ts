export const LOGO_IMAGE = '/images/triple2x.png'
export const AWARD_IMAGE_GOOGLE = '/images/play-store2x.png'
export const AWARD_IMAGE_APPLE = '/images/badge-apple4x.png'
