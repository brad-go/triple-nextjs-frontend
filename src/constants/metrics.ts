export const MetricItemList = [
  {
    metric: 700,
    unit: '만 명의',
    subject: '여행자',
  },
  {
    metric: 100,
    unit: '만 개의',
    subject: '여행 리뷰',
  },
  {
    metric: 460,
    unit: '만 개의',
    subject: '여행 일정',
  },
]
