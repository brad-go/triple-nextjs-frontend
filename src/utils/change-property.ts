export const changePropertyCamelToKebab = (property: string) => {
  const capital = property.match(/[A-Z]/g)

  if (capital) {
    const capitalIndex = property.indexOf(capital[0])
    const firstWord = property.slice(0, capitalIndex)
    const secondWord = capital[0]
      .toLocaleLowerCase()
      .concat(property.slice(capitalIndex + 1))

    property = `${firstWord}-${secondWord}`
  }

  return property
}
