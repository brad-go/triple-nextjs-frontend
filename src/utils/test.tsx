import { PropsWithChildren, ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import { theme } from 'styles/theme'

function Provider<T>({ children }: PropsWithChildren<T>) {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: Provider, ...options })

export { customRender as render }
