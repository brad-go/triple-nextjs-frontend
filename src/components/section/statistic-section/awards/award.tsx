import { forwardRef } from 'react'
import styled, { CSSProperties } from 'styled-components'

import { AwardList, AWARD_LIST } from 'constants/awards'
import { UlRefType } from 'types/section'

interface AwardProps {
  awards?: AwardList[]
  style?: CSSProperties
}

const AwardItem = styled.li<{ src: string }>`
  display: inline-block;
  background-image: ${({ src }) => `url(${src})`};
  background-position: left top;
  background-repeat: no-repeat;
  font-family: sans-serif;
  font-weight: bold;
  color: ${({ theme }) => theme.color.gray800};

  @media ${({ theme }) => theme.device.pc} {
    height: 54px;
    margin-right: 39px;
    padding: 5px 0px 5px 62px;
    background-size: 54px 54px;
    font-size: 14px;
    line-height: 22px;
  }
  @media ${({ theme }) => theme.device.mobile} {
    padding: 4px 0px 4px 45px;
    background-size: 39px 39px;
    font-size: 10px;
    line-height: 15px;

    &:first-child {
      margin-right: 28.5px;
    }
  }
`

const AwardContainer = styled.ul`
  white-space: nowrap;

  @media ${({ theme }) => theme.device.pc} {
    margin: 50px 0px 140px 623px;
  }

  @media ${({ theme }) => theme.device.mobile} {
    width: 300px;
    margin: 50px auto 80px;
  }
`

function Awards({ awards = AWARD_LIST, style }: AwardProps, ref: UlRefType) {
  return (
    <AwardContainer ref={ref} style={style}>
      {awards.map(({ id, platform, award, image }) => (
        <AwardItem key={id} src={image}>
          {platform}
          <br />
          {award}
        </AwardItem>
      ))}
    </AwardContainer>
  )
}

const Award = forwardRef<HTMLUListElement, AwardProps>(Awards)

export default Award
