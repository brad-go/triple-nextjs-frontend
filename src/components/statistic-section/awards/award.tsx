import styled from 'styled-components'

import { AwardList, AWARD_LIST } from 'constants/awards'

interface AwardProps {
  awards?: AwardList[]
}

const AwardItem = styled.li<{ src: string }>`
  display: inline-block;
  background-image: ${({ src }) => `url(${src})`};
  background-position: left top;
  background-repeat: no-repeat;
  font-family: sans-serif;
  font-weight: bold;
  color: ${({ theme }) => theme.color.gray800};

  @media ${({ theme }) => theme.device.pc} {
    margin-right: 39px;
    padding: 5px 0px 5px 62px;
    background-size: 54px 54px;
    font-size: 14px;
    line-height: 22px;
  }
  @media ${({ theme }) => theme.device.mobile} {
    padding: 4px 0px 4px 45px;
    background-size: 39px 39px;
    font-size: 10px;
    line-height: 15px;

    &:first-child {
      margin-right: 28.5px;
    }
  }
`

const AwardContainer = styled.ul`
  @media ${({ theme }) => theme.device.pc} {
    margin: 50px 0px 140px 623px;
  }

  @media ${({ theme }) => theme.device.mobile} {
    width: 300px;
    margin: 50px auto 80px;
  }
`

function Award({ awards = AWARD_LIST }: AwardProps) {
  return (
    <AwardContainer>
      {awards.map(({ id, platform, award, image }) => (
        <AwardItem key={id} src={image}>
          {platform}
          <br />
          {award}
        </AwardItem>
      ))}
    </AwardContainer>
  )
}

export default Award
