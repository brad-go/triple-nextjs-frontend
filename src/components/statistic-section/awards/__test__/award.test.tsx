import Award from '../award'

import { render } from 'utils/test'
import { AWARD_LIST } from 'constants/awards'

describe('Award', () => {
  it('should be rendered', () => {
    const { container } = render(<Award />)

    expect(container).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should be rendered', () => {
    const { getAllByRole } = render(<Award awards={AWARD_LIST} />)

    getAllByRole('listitem')
  })
})
