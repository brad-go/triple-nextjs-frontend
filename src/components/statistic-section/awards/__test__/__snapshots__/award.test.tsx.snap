// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Award should be rendered 1`] = `
.c1 {
  display: inline-block;
  background-image: url(/images/play-store2x.png);
  background-position: left top;
  background-repeat: no-repeat;
  font-family: sans-serif;
  font-weight: bold;
  color: #3a3a3acc;
}

.c2 {
  display: inline-block;
  background-image: url(/images/badge-apple4x.png);
  background-position: left top;
  background-repeat: no-repeat;
  font-family: sans-serif;
  font-weight: bold;
  color: #3a3a3acc;
}

@media screen and (min-width:1142px) {
  .c1 {
    margin-right: 39px;
    padding: 5px 0px 5px 62px;
    background-size: 54px 54px;
    font-size: 14px;
    line-height: 22px;
  }
}

@media screen and (max-width:1141px) {
  .c1 {
    padding: 4px 0px 4px 45px;
    background-size: 39px 39px;
    font-size: 10px;
    line-height: 15px;
  }

  .c1:first-child {
    margin-right: 28.5px;
  }
}

@media screen and (min-width:1142px) {
  .c2 {
    margin-right: 39px;
    padding: 5px 0px 5px 62px;
    background-size: 54px 54px;
    font-size: 14px;
    line-height: 22px;
  }
}

@media screen and (max-width:1141px) {
  .c2 {
    padding: 4px 0px 4px 45px;
    background-size: 39px 39px;
    font-size: 10px;
    line-height: 15px;
  }

  .c2:first-child {
    margin-right: 28.5px;
  }
}

@media screen and (min-width:1142px) {
  .c0 {
    margin: 50px 0px 140px 623px;
  }
}

@media screen and (max-width:1141px) {
  .c0 {
    width: 300px;
    margin: 50px auto 80px;
  }
}

<div>
  <ul
    class="c0"
  >
    <li
      class="c1"
      src="/images/play-store2x.png"
    >
      2018 구글 플레이스토어
      <br />
      올해의 앱 최우수상 수상
    </li>
    <li
      class="c2"
      src="/images/badge-apple4x.png"
    >
      2018 애플 앱스토어
      <br />
      오늘의 여행앱 선정
    </li>
  </ul>
</div>
`;
